src/test/java/fi/aalto/cs/apluscourses/intellij/actions/ImportProjectSettingsActionTest.java

```diff
@@ -0,0 +1,119 @@
+package fi.aalto.cs.apluscourses.intellij.actions;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.model.Course;
+import fi.aalto.cs.apluscourses.presentation.CourseViewModel;
+import fi.aalto.cs.apluscourses.presentation.MainViewModel;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ImportProjectSettingsActionTest {
+
+
+  private Project project;
+  private MainViewModel mainViewModel;
+  private AnActionEvent anActionEvent;
+
+  /**
+   * Called before each test method call.  Initializes private fields.
+   */
+  @Before
+  public void createMockObjects() throws MalformedURLException {
```

Comment:
there are some minor complaints on "duplication" from Sonar, but they can easily be pushed to that refactoring stage

Reply:
Sounds good, I'll add a note of this